doc/manual/mathematical-operations.rst

```diff
@@ -454,6 +454,7 @@ Function                     Description
 :func:`cld(x,y) <cld>`       ceiling division; quotient rounded towards ``+Inf``
 :func:`rem(x,y) <rem>`       remainder; satisfies ``x == div(x,y)*y + rem(x,y)``; sign matches ``x``
 :func:`mod(x,y) <mod>`       modulus; satisfies ``x == fld(x,y)*y + mod(x,y)``; sign matches ``y``
+:func:`mod1(x,y) <mod1>`     modulus with offset 1; gives a result ``r`` such that ``mod(r, y)==mod(x,y)``; sign matches ``y``
```

Comment:
It's not quite a definition since it doesn't define `r`; the condition you quote (which is quite elegant, I didn't realize this before) allows for many values of `r`. If you combine it with the condition `0 < r <= y` (for positive `y`) etc. you'll have a complete definition. You might even want to update the existing documentation of `mod1` with this.